damnit/gui/main_window.py

```diff
@@ -1094,17 +1095,78 @@ def drawControl(self, element, option, painter, widget=None):
 
 
 class LogViewWindow(QtWidgets.QMainWindow):
-    def __init__(self, file_path: Path, parent=None):
+    def __init__(self, file_path: Path, parent=None, polling_interval_ms: int = 1500):
         super().__init__(parent)
         self.file_path = file_path
-        self.text_edit = QtWidgets.QPlainTextEdit(file_path.read_text())
+        self.text_edit = QtWidgets.QPlainTextEdit()
         self.text_edit.setReadOnly(True)
         font = self.text_edit.document().defaultFont()
         font.setFamily('monospace')
         self.text_edit.document().setDefaultFont(font)
         self.setCentralWidget(self.text_edit)
         self.resize(1000, 800)
 
+        # Read initial file content and keep track of the file size to only append new content
+        self.text_edit.setPlainText(self.file_path.read_text(encoding='utf-8', errors='replace'))
+        stat_res = self.file_path.stat()
+        self._last_mtime = stat_res.st_mtime
+        self._last_size = stat_res.st_size
+
+        # Poll for file changes using a timer
+        self._timer = QTimer(self)
+        self._timer.setInterval(polling_interval_ms)
+        self._timer.timeout.connect(self._check_log_file_update)
+        self._timer.start()
+
+    def _check_log_file_update(self):
+        """Poll the file for changes in mtime or size."""
+        try:
+            stat_res = self.file_path.stat()
+        except FileNotFoundError:
+            # File might have been deleted
+            if self._last_size is not None:  # Avoid repeated logging
+                log.warning(f"Log file {self.file_path} not found.", exc_info=True)
+                self.text_edit.setPlainText(f"[Log file {self.file_path} not found or deleted]")
```

Comment:
I might just leave the previous contents visible in this scernario, but I don't feel strongly about it.